src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.hellbender.engine.spark.datasources;
+
+
+import htsjdk.samtools.SAMFileHeader;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.spark.SparkTestUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadCoordinateComparator;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+
+public class ReadsSparkSinkUnitTest extends BaseTest {
+    @DataProvider(name = "loadReads")
+    public Object[][] loadReads() {
+        String dir = "src/test/resources/org/broadinstitute/hellbender/tools/BQSR/";
+        return new Object[][]{
+                {dir + "HiSeq.1mb.1RG.2k_lines.alternate.bam", "test1.bam"},
+                {dir + "expected.HiSeq.1mb.1RG.2k_lines.bqsr.DIQ.alternate.bam", "test2.bam"},
```

Comment:
`Use `BaseTest.createTempFile()` to create the output files, and use a test-specific prefix instead of generic names like `test1.bam`